pkg/engine/renderer/dependencies_test.go

```diff
@@ -0,0 +1,173 @@
+package renderer
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"gotest.tools/assert"
+	v1 "k8s.io/api/apps/v1"
+	"k8s.io/api/batch/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/kudobuilder/kudo/pkg/util/kudo"
+)
+
+func TestGetResources(t *testing.T) {
+	cm := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "configmap",
+			Namespace:   "namespace",
+			Annotations: map[string]string{},
+		},
+		Data: map[string]string{
+			"key": "value",
+		},
+	}
+	cmString, _ := runtime.Encode(unstructured.UnstructuredJSONScheme, &cm)
+	cm.Annotations[kudo.LastAppliedConfigAnnotation] = string(cmString)
+
+	cmUnstructuredData, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(&cm)
+	cmUnstructured := unstructured.Unstructured{Object: cmUnstructuredData}
+
+	tests := []struct {
+		name        string
+		taskObjects []*unstructured.Unstructured
+		client      client.Client
+	}{
+		{name: "from api server", taskObjects: []*unstructured.Unstructured{}, client: fake.NewFakeClientWithScheme(scheme.Scheme, &cm)},
+		{name: "from task objects", taskObjects: []*unstructured.Unstructured{&cmUnstructured}, client: fake.NewFakeClientWithScheme(scheme.Scheme)},
+	}
+
+	for _, test := range tests {
```

Comment:
testing nit: if you use:
```
for _, tt := range tests {
	tt := tt
	t.Run(tt.name, func(t *testing.T) {
               ...
         }
```
you'll have each individual test case (with name) in the console output *and* in Intellij overview. Much easier to see which test case failed and in Intellij you can re-run only failed ones ;) Same below.

Reply:
Oh, neat. I like that.